services/provider/server/server.go

```diff
@@ -814,15 +825,15 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 		return nil, status.Errorf(codes.Internal, "Failed to get storageConsumer resource: %v", err)
 	}
 
-	channelName, err := s.getOCSSubscriptionChannel(ctx)
+	subscriptionSpec, err := s.getOCSSubscriptionSpec(ctx)
```

Comment:
We should not create an entire spec to get the channel name. preferably have a func to get the channel name then use it here and when creating the spec.